rfhub2/cli/rfhub_importer.py

```diff
@@ -0,0 +1,175 @@
+from pathlib import Path
+import re
+from robot.libdocpkg import LibraryDocumentation
+import robot.libraries
+from typing import Tuple
+
+from .api_client import Client
+from rfhub2.model import Collection, Keyword
+
+
+RESOURCE_PATTERNS = {'.robot', '.txt', '.tsv', '.resource'}
+ALL_PATTERNS = (RESOURCE_PATTERNS | {'.xml', '.py'})
+EXCLUDED_LIBRARIES = {'remote.py', 'reserved.py', 'dialogs.py', 'dialogs_jy.py', 'dialogs_py.py', 'dialogs_ipy.py'}
+INIT_FILES = {'__init__.txt', '__init__.robot', '__init__.html', '__init__.tsv'}
+
+
+class RfhubImporter(object):
+
+    def __init__(self, paths: Tuple[Path, ...], no_installed_keywords: bool, client: Client) -> None:
+        self.paths = paths
+        self.no_installed_keywords = no_installed_keywords
+        self.client = client
+
+    def delete_collections(self) -> None:
```

Comment:
@pbylicki I do lack of idea how to unit test that method, as well as `add_collections`, `_traverse_paths` and `add`. Could You provide some ideas? I really want to keep high coverage :)

Reply:
@MaciejWiczk As for the http request parts, see my other comment on unit testing them. As for file system operations with pathlib, you could also mock the methods from this library to return expected objects. There are some nice tutorials on the internet how to use built-in unittest.mock package, but if you want I can prepare test as an example for you

Reply:
@MaciejWiczk As for testing methods relying on pathlib methods, here is simple example of how you could do it:

```suggestion
    def test_traverse_paths(self):
        def mock_iterdir():
            return [self.fixture_path / 'SingleClassLib' / 'SingleClassLib.py']
        path = Mock()
        path.iterdir = mock_iterdir
        self.rfhub_importer._traverse_paths(path)`
```
You can provide mocked iterdir method returning instances of Path objects matching conditions that you want to test, or if you do not have existing files you could also returned mocked Path object in that list.

Current implementation of this methods is not really easy for testing because they do not return anything, and the flow is not very straightforward. What I would do is to change the implementation of the flow to be something like this:
1. method to get sequence of valid library paths based on input parameters
2. method to map path to Collection object (containing list of Keyword objects)
3. method to send collection and its keywords using client (it can also return not None response for some final report of how many were sent etc. but it is not necessary in this PR)

In this way testing this three methods should be very easy, with method 1 relying on mocked iterdir like in example above, method 2 requiring no mocks and method 3 relying on mock of client's add collection/keyword methods. I think that struggling to create unit tests without  such refactor has no point.

Reply:
@pbylicki Ok, so I will start with rafactoring rfhub_importer.py according to points 1-3. regarding not None response, I do print out library name and how many keywords were added:
```
BuiltIn library with 104 keywords loaded.
Collections library with 43 keywords loaded.
DateTime library with 8 keywords loaded.
Easter library with 1 keywords loaded.
```
Do You have something else in mind?

Reply:
@MaciejWiczk I thought about displaying total number of loaded collections and keywords, but again, it does not need to be part of this PR.
Yes, I would not expect a scenario that collection objects would be too big to store in memory. You can also use yielding results from this method one by one instead of returning whole list if you want.

Reply:
@pbylicki one more question regarding point 2. Can collection with keywords be posted as one, or should keywords be posted separately as they are now?

Reply:
API does not allow posting collection with nested keywords, but you can pass collection with keywords to method which then will send collection first, and all its keywords after that

Reply:
Ok, that's cool. I will do exactly as You say. I;m almost one with nit testing, hopefully it will work as I expect in e2e ;)